### PR TITLE
Adds False Walls and False Reinforced Walls

### PR DESCRIPTION
**Pull request by Robustin**
Description:

False walls! The cornerstone of any stealthy antagonist. The behavior is identical to how false walls functioned on /tg/code, including the massive headache of getting it to smooth with nearby walls.

This also resolves a bug where GeneralDoorAnimator.cs would break any door that used less than 6 frames for animation and it adds "Non-Automatic" door behavior that can be used for mineral doors and other unpowered doors that don't simply open from being bumped. It also adds a "isHighlightable" object trait that prevents highlighting of objects when false. 

_Now with a branch that isn't FUBAR_

This is a bounty (#2358).

Remaining Work:

- None atm, awaiting any further feedback.

### CODE REVIEW

**Comment by Evan-R-Rogers** [comment]
Please tell me you added it to the tile palette for doors?

**Comment by Robustin** [comment]
I asked in the last PR for clarification on that. It has its sprites kept with the other "General Door Animation" doors.

**Comment by Evan-R-Rogers** [comment]
This link should help ya: [bwoink](https://unitystation.github.io/unitystation/development/How-to-Map/#edit-a-palette)

**Comment by corp-0** [comment]
that commit history, I felt the pain.

**Comment by Robustin** [comment]
@Evan-R-Rogers 

From what I've been able to deduce, it wouldn't be proper for me to add this to the Tile Palette. It's classified as a "door" and we do not seem to use the tile palette for doors (the doors category on the tile palette is literally empty). It's a prefab object like all the other doors.

@corp-0 

Yea, it's uh, oof. I'm still re-acclimating to Git and learning Unity, the two do not appear to play very nicely with one another. Thankfully, it's been getting a little easier to avoid some of these commit issues.

File: UnityProject/Assets/Scripts/Doors/DoorController.cs [hunk 1]
```diff
@@ -292,6 +298,14 @@ private void SyncIsWelded(bool _wasWelded, bool _isWelded)
...
 			}
 		}
+		public void ServerDisassemble(HandApply interaction)
+		{
+			Vector3Int position = Vector3Int.RoundToInt(transform.localPosition);
+			// tileChangeManager.RemoveTile(position, LayerType.Walls); Disabled until I can figure out
+			// tileChangeManager.SubsystemManager.UpdateAt(position); why removing a passable wall deletes atmos
+			Spawn.ServerPrefab(CommonPrefabs.Instance.Metal, registerTile.WorldPositionServer, count: 4);
+			GetComponent<CustomNetTransform>().DisappearFromWorldServer();
+		}
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Doors/DoorController.cs]: DisappearFromWorld server doesn't despawn prefab but moves it to hiddenpos forever. that and #1902 probably make that nasty no pressure effect when combined.
use this:
```suggestion
        public void ServerDisassemble(HandApply interaction)
        {
            tileChangeManager.RemoveTile(registerTile.LocalPositionServer, LayerType.Walls);
            Spawn.ServerPrefab(CommonPrefabs.Instance.Metal, registerTile.WorldPositionServer, count: 4);
            Despawn.ServerSingle(gameObject);
        }
```
but! keep in mind that if false wall will be destroyed due to shooting, `            tileChangeManager.RemoveTile(registerTile.LocalPositionServer, LayerType.Walls);` won't be called, see https://github.com/unitystation/unitystation/blob/26339d2153e59fbed87b06fe43bf438a8ab1ccb1/UnityProject/Assets/Scripts/Tilemaps/Behaviours/Objects/RegisterDoor.cs#L52-L57

**Review by fomalsd — CHANGES_REQUESTED** [review]
looks like metafiles are gitignored in resources/icons, since all textures are supposed to be stored outside of Resources folder anyway, but doors might be a special case unaccounted for. yet that metafile is crucial for this whole thing to work:
[metal_fwall_closing.png.meta.txt](https://github.com/unitystation/unitystation/files/4263207/metal_fwall_closing.png.meta.txt)
other than that it's great

**Comment by fomalsd** [comment]
also could you please readd this script I deleted earlier today (to Scripts/Editor/Tools)
https://gist.github.com/fomalsd/90d5916e1375c2771669405c9c14e68d

**Comment by Robustin** [comment]
@fomalsd 

The last commit addressed the remaining issues. 

I re-added the tool in #3394. I pulled onto a clean updated branch and everything appeared to be working great. Was there any other problems you noticed?

File: UnityProject/Assets/Scripts/Objects/Construction/Girder.cs [hunk 2]
```diff
@@ -186,5 +206,25 @@ private void ReinforceGirder(HandApply interaction)
 		Despawn.ServerSingle(gameObject);
 	}
 
+	[Server]
+	private void ConstructFalseWall(HandApply interaction)
+	{
+		GameObject theWall = Spawn.ServerPrefab(FalseWall, SpawnDestination.At(gameObject)).GameObject;
+		DoorController doorController = theWall.GetComponent<DoorController>();
+		tileChangeManager.UpdateTile(Vector3Int.RoundToInt(transform.localPosition), falseTile);
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Objects/Construction/Girder.cs]:
```suggestion
		tileChangeManager.UpdateTile(registerObject.LocalPositionServer, falseTile);
```

File: UnityProject/Assets/Scripts/Objects/Construction/Girder.cs [hunk 3]
```diff
@@ -186,5 +206,25 @@ private void ReinforceGirder(HandApply interaction)
...
+		Despawn.ServerSingle(gameObject);
+		doorController.ServerTryClose();
+	}
 
+	[Server]
+	private void ConstructReinforcedFalseWall(HandApply interaction)
+	{
+		GameObject theWall = Spawn.ServerPrefab(FalseReinforcedWall, SpawnDestination.At(gameObject)).GameObject;
+		DoorController doorController = theWall.GetComponent<DoorController>();
+		tileChangeManager.UpdateTile(Vector3Int.RoundToInt(transform.localPosition), falseTile);
```
Comment (fomalsd) [inline UnityProject/Assets/Scripts/Objects/Construction/Girder.cs]:
```suggestion
		tileChangeManager.UpdateTile(registerObject.LocalPositionServer, falseTile);
```

**Review by fomalsd — COMMENTED** [review]
use of transform.position is discouraged since it can be inaccurate. always try to use registerTile position for objects
will playtest again later today and see if I can find any quirks

**Review by fomalsd — COMMENTED** [review]
it's all good gameplay wise, just apply suggested changes and I'll merge this

**Comment by Robustin** [comment]
Oh you can commit suggestions now, neat.

**Review by fomalsd — APPROVED** [review]
awesome

**Comment by DooblyNoobly** [comment]
@Robustin can you pm me your paypal email on discord pls

**Pull request merged**